tools/bowtie2.wdl

```diff
@@ -0,0 +1,410 @@
+        no_discordant: "By default, bowtie2 looks for discordant alignments if it cannot find any concordant alignments. A discordant alignment is an alignment where both mates align uniquely, but that does not satisfy the paired-end constraints. This option disables that behavior."
+        dovetail: {
+            description: "If the mates 'dovetail', that is if one mate alignment extends past the beginning of the other such that the wrong mate begins upstream, consider that to be concordant.",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#mates-can-overlap-contain-or-dovetail-each-other",
+        }
+        no_contain: {
+            description: "If one mate alignment contains the other, consider that to be non-concordant.",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#mates-can-overlap-contain-or-dovetail-each-other",
+        }
+        no_overlap: {
+            description: "If one mate alignment overlaps the other at all, consider that to be non-concordant.",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#mates-can-overlap-contain-or-dovetail-each-other",
+        }
+        time: "Print the wall-clock time required to load the index files and align the reads to STDERR."
+        write_unpaired_unaligned: "Write unpaired reads that didn't align to a file in gzipped FASTQ format. These reads correspond to the SAM records with the FLAGS 0x4 bit set and neither the 0x40 nor 0x80 bits set."
+        write_unpaired_aligned: "Write unpaired reads that aligned at least once to a file in gzipped FASTQ format. These reads correspond to the SAM records with the FLAGS 0x4, 0x40, and 0x80 bits unset."
+        write_paired_discordant: "Write pairs that didn't align concordantly to a file in gzipped FASTQ format. These reads correspond to the SAM records with the FLAGS 0x4 bit set and either the 0x40 or 0x80 bit set (depending on whether it's mate #1 or #2)."
+        write_paired_concordant: "Write pairs that aligned concordantly at least once to a file in gzipped FASTQ format. These reads correspond to the SAM records with the FLAGS 0x4 bit unset and either the 0x40 or 0x80 bit set (depending on whether it's mate #1 or #2)."
+        quiet: "Print nothing besides alignments and serious errors."
+        metrics_file: "Write bowtie2 alignment metrics to file. Useful for debugging."
+        metrics_stderr: "Write  bowtie2 alignment metrics to STDERR. Not mutually exclusive with `metrics_file`."
+        metrics_interval: "Report internal counters & metrics every N seconds. Only used if `metrics_file` and/or `metrics_stderr` is specified."
+        no_unal: "Suppress SAM records for unaligned reads."
+        no_head: "Suppress header lines, i.e. lines starting with @."
+        no_sq: "Suppress @SQ header lines."
```

Comment:
Are these options just allowing malformed SAM files to be created? Should this be hidden?

Reply:
I'd be fine disallowing them. If a user wants to do that, I think they should be forced to explicitly remove those lines separately.

Reply:
That said I'm guessing there are some old tools that don't support SAM headers that tread bowtie output.